src/modals/cluster-upgrade/ClusterUpgradeWizard.js

```diff
@@ -69,7 +69,7 @@ const ClusterUpgradeWizard = ({
       clusterId: cluster.id,
       clusterName: cluster.name,
 
-      hostNames: clusterHosts.filter(host => selectedHostIds[host.id]).map(host => hostIdToName[host.id]),
+      hostNames: selectedHostIds.map(hostId => hostIdToName[hostId]),
```

Comment:
1. In `onHostSelectionChange` we check if the hosts exist. Should we check it here too  i.e.  with `filter(Boolean)` ?
2. Can `hostIdToName`be calculated as part of render phase?

Reply:
> 1. In `onHostSelectionChange` we check if the hosts exist. Should we check it here too  i.e.  with `filter(Boolean)` ?

I don't think that is necessary since only `onHostSelectionChange` actually sets the `selectedHostIds`.

> 
>   2. Can `hostIdToName`be calculated as part of render phase?

Humm, I'd rather do it in the event handler functions to be safe.